Libraries/react-native/react-native.js

```diff
@@ -10,14 +10,18 @@
  */
 'use strict';
 
+var React = require('React');
+
 // Export React, plus some native additions.
 //
 // The use of Object.create/assign is to work around a Flow bug (#6560135).
 // Once that is fixed, change this back to
 //
-//   var ReactNative = {...require('React'), /* additions */}
+//   var ReactNative = {...React, /* additions */}
 //
-var ReactNative = Object.assign(Object.create(require('React')), {
+var ReactNative = Object.assign(Object.create(React), {
+  React,
```

Comment:
There are many issues with

```javascript
var React = require('react-native');
var Navigator = require('react-native-navigator');
var Vibration = require('react-native-vibration');
```

The first one is that we're going to run into naming conflicts with npm. Someone else will steal a name that we want and we're going to either reach out and negotiate or use a less ideal one.

Then, you're running into crazy dependency issues where you have different versions of all of those plugins and conflicts. We really want to ship one bundle where everything is in sync for the core components.

One idea I had is:

```javascript
var React = require('react-native/react');
var Navigator = require('react-native/navigator');
var Vibration = require('react-native/vibration');
```

Everything is bundled together, no namespace clash and we don't bundle things that we don't use